### PR TITLE
[prometheus-adapter] cert-manager support

### PR DESCRIPTION
**Pull request by velothump**
Signed-off-by: Adam Graves <adam.graves85@gmail.com>

#### What this PR does / why we need it:

Adds cert-manager support to the prometheus-adapter apiservice.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-adapter/templates/external-metrics-apiservice.yaml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {{- if .Values.rules.external }}
-apiVersion: apiregistration.k8s.io/v1beta1
+apiVersion: apiregistration.k8s.io/v1
```
Comment (monotek) [inline charts/prometheus-adapter/templates/external-metrics-apiservice.yaml:2]: does introduce breaking changes to older versions?

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/templates/external-metrics-apiservice.yaml (hunk 1 above)
Comment (velothump) [reply charts/prometheus-adapter/templates/external-metrics-apiservice.yaml]: You're right, I've reverted it back now :)

Comment (monotek) [reply charts/prometheus-adapter/templates/external-metrics-apiservice.yaml]: You could add an if to check for the k8s version and use the api version available.

Example: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/_helpers.tpl#L183-L186

Comment (velothump) [reply charts/prometheus-adapter/templates/external-metrics-apiservice.yaml]: I thought about that, but there's already a PR open for it: https://github.com/prometheus-community/helm-charts/pull/463. I thought it best just to keep this PR as simple as possible for now.

**Review by monotek — APPROVED** [review]

**Review by monotek — APPROVED** [review]

File: charts/prometheus-adapter/templates/certmanager.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    secretName: {{ template "k8s-prometheus-adapter.fullname" . }}-root-cert
+---
+# Finally, generate a serving certificate for the apiservices to use
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ template "k8s-prometheus-adapter.fullname" . }}-cert
+spec:
+  secretName: {{ template "k8s-prometheus-adapter.fullname" . }}
+  duration: 8760h # 1y
```
Comment (hectorj2f) [inline charts/prometheus-adapter/templates/certmanager.yaml:41]: This should be configurable.

File: charts/prometheus-adapter/templates/certmanager.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  selfSigned: {}
+---
+# Generate a CA Certificate used to sign certificates for the webhook
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ template "k8s-prometheus-adapter.fullname" . }}-root-cert
+spec:
+  secretName: {{ template "k8s-prometheus-adapter.fullname" . }}-root-cert
+  duration: 43800h # 5y
```
Comment (hectorj2f) [inline charts/prometheus-adapter/templates/certmanager.yaml:19]: This should be configurable.

**Review by monotek — CHANGES_REQUESTED** [review]
Please implement @hectorj2f suggestions.

**Comment by velothump** [comment]
@monotek Updated the PR now to make those configurable.

File: charts/prometheus-adapter/templates/certmanager.yaml (hunk 3 above)
Comment (velothump) [reply charts/prometheus-adapter/templates/certmanager.yaml]: This has now been moved to the values.yaml file

File: charts/prometheus-adapter/templates/certmanager.yaml (hunk 2 above)
Comment (velothump) [reply charts/prometheus-adapter/templates/certmanager.yaml]: This has now been moved to the values.yaml file

**Review by hectorj2f — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**